src/temp_sensor.rs

```diff
@@ -0,0 +1,149 @@
+#[cfg(not(any(esp32)))]
+use esp_idf_sys::{
+    esp, soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_DEFAULT,
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST,
+    soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL,
+    temperature_sensor_clk_src_t, temperature_sensor_config_t, temperature_sensor_disable,
+    temperature_sensor_enable, temperature_sensor_get_celsius, temperature_sensor_handle_t,
+    temperature_sensor_install, temperature_sensor_uninstall, EspError,
+};
+
+// -- TemperatureSensorClockSource --
+
+#[derive(Copy, Clone)]
+#[cfg(not(any(esp32)))]
+/// Rust translation of `temperature_sensor_clk_src_t`
+pub enum TemperatureSensorClockSource {
+    Default,
+    XTAL,
+    RcFast,
+}
+
+#[cfg(not(any(esp32)))]
+impl From<TemperatureSensorClockSource> for temperature_sensor_clk_src_t {
+    fn from(value: TemperatureSensorClockSource) -> Self {
+        match value {
+            TemperatureSensorClockSource::XTAL => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL
+            }
+            TemperatureSensorClockSource::RcFast => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST
+            }
+            TemperatureSensorClockSource::Default => {
+                soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_DEFAULT
+            }
+        }
+    }
+}
+
+#[cfg(not(any(esp32)))]
+impl From<temperature_sensor_clk_src_t> for TemperatureSensorClockSource {
+    fn from(value: temperature_sensor_clk_src_t) -> Self {
+        match value {
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_XTAL => Self::XTAL,
+            #[allow(non_upper_case_globals)]
+            soc_periph_temperature_sensor_clk_src_t_TEMPERATURE_SENSOR_CLK_SRC_RC_FAST => {
+                Self::RcFast
+            }
+            // TODO: Perhaps the default value should be mapped explicitly
+            // and all other (u32) values should cause a failure
+            _ => Self::Default,
+        }
+    }
+}
+
+// -- TemperatureSensorConfig --
+
+#[cfg(not(any(esp32)))]
+#[derive(Copy, Clone)]
```

Comment:
- Remove the `Copy` derivation
- Add a `Debug` derivation and why not `PartialEq`, `Eq` and `Hash`